chunks/generative_models.ipynb

```diff
@@ -0,0 +1,397 @@
+    "\n",
+    "* You may have encountered some of these routines as \"random number generators\"\n",
+    "\n",
+    "* Sampling from a PDF means generating random numbers from that distribution"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Choosing the input model parameters\n",
+    "\n",
+    "* When testing, we often want to assert a set of input model parameters $\\theta$, and then see what they produce\n",
+    "\n",
+    "* Testing at large scale might involve generating many datasets, with different inputs. In this case we might want to sample from a *plausible distribution of input parameters*\n",
+    "\n",
+    "* In practice: choose a particular standard probability distribution for $\\theta$ and sample from it"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Deterministic relationships\n",
+    "\n",
+    "* Often, our model provides a deterministic relationship between parameters: if you know $\\theta$, then you know $\\mu$\n",
+    "\n",
+    "* In our X-ray image case, $\\theta$ could be the set of parameters that describe the gas temperature and density profiles of a spehrically symmetric cluster of galaxies with known centroid position\n",
+    "\n",
+    "* $\\mu(\\theta)$ would then be some complicated function that took the cluster parameters $\\theta$ as input, and predicted the expected counts at any pixel position."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Probabilistic Graphical Models\n",
+    "\n",
+    "* This procedure for simulating a mock X-ray image dataset can be usefully illustrated with a _directed acyclic graph_ called a \"Probabilistic Graphical Model\"\n",
+    "\n",
+    "* In the present context these provide something like a \"flow diagram\" showing how we might draw a sample mock dataset from our model\n",
+    "\n",
+    "* Let's look at the PGM for a simple X-ray image simulation \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## One pixel, fixed inputs\n",
+    "\n",
+    "<img src=\"../graphics/pgms_one_pixel_input_fixed.png\">"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## PGM pieces\n",
+    "\n",
+    "* Each _node_ in the graph represents a PDF, for the variable labeled inside it\n",
+    "\n",
+    "* Each _edge_ in the graph represents a conditional dependence\n",
+    "\n",
+    "* Deterministic relationships are indicated by \"fixed\" variables represented by solid points\n",
```

Comment:
If we're going to use this convention, I should re-make my PGMs in the Bayes lecture. I always used points for things that are actually constant, rather than (more weakly) deterministic.